src/context/PortfolioProvider/PortfolioContext.tsx

```diff
@@ -171,6 +175,46 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
     })()
   }, [assetsById, chainAdapter, dispatch, wallet])
 
+  const txIds = useSelector(selectTxIds)
+  const txsById = useSelector(selectTxs)
+  const txHistoryStatus = useSelector(selectTxHistoryStatus)
+
+  /**
+   * portfolio refetch on new tx logic
+   */
+  useEffect(() => {
+    // we only want to refetch portfolio if a new tx comes in after we're finished loading
+    if (txHistoryStatus !== 'loaded') return
+    // don't fire with nothing connected
+    if (isEmpty(accountSpecifiers)) return
+    // grab the most recent txId
+    const txId = head(txIds)!
+    // grab the actual tx
+    const tx = txsById[txId]
+    // always wear protection, or don't it's your choice really
+    if (!tx) return
+    // the chain the tx came from
+    const txRelatedAccount = txId.split('-')[0]
+    // only refetch accounts for this tx
+    const accountSpecifierMap = accountSpecifiersList.reduce((acc, cur) => {
+      const [chainId, accountSpecifier] = Object.entries(cur)[0]
+      const accountId = chainId + ':' + accountSpecifier
+      // NOTE: in case of ethereum on Portis wallet, accountId contains both
```

Comment:
this is not a good place for a portis specific fix - this should be ticketed and addressed in `useWallet`

Reply:
here: #1376